pandas/core/tools/datetimes.py

```diff
@@ -517,6 +487,93 @@ def _array_strptime_with_fallback(
     return _box_as_indexlike(result, utc=utc, name=name)
 
 
+def _to_datetime_with_format(
```

Comment:
where is this called? (now)

Reply:
this is basically cut/paste/de-dented from _convert_listlike_datetimes

Reply:
woops, did the copy/paste but forgot to actually use the new function.  updated